### PR TITLE
Adding support for installing roles with ansible-galaxy for ansible provisioner

### PR DESCRIPTION
**Pull request by rubentsirunyan**
This is an attempt to solve [this ](https://github.com/hashicorp/packer/issues/3301) issue for Ansible provisioner too.

This is my first experience with go, so I am looking forward for some comments and advices :)

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
:+1:

File: provisioner/ansible/provisioner.go [hunk 1]
```diff
@@ -100,6 +104,14 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 		p.config.Command = "ansible-playbook"
 	}
 
+	if p.config.GalaxyCommand == "" {
+		p.config.GalaxyCommand = "ansible-galaxy"
+	}
+
+	if p.config.RolesDir == "" {
+		p.config.RolesDir = "~/.ansible/roles"
+	}
```
[the remark this answers is not in the document]
Comment (rubentsirunyan) [reply provisioner/ansible/provisioner.go]: Done.

**Comment by rubentsirunyan** [comment]
What is the process of resolving the conflicts? Should I resolve it?

**Comment by rickard-von-essen** [comment]
Try something like:

```
git checkout master; git pull
git checkout ansible-galaxy-support; git rebase master
# fix conflicts and follow the instructions in the rebase and "git status" 
# when you are done verify with
git diff HEAD master
# if everything is fine
git push --force-with-lease <remote> ansible-galaxy-support
```

**Comment by rubentsirunyan** [comment]
Conflicts are resolved.
I think WIP label can already be removed.
However I think that something like a `tempt_roles_path` could be another useful parameter. If the parameter is specified, a temporary directory will be created for installing the roles in it and then deleted after the playbook is run. What are your thoughts on this? Maybe I can add it in another PR later.

**Comment by SwampDragons** [comment]
That does sound useful and does sound like something for a different PR 👍

**Pull request merged**